### PR TITLE
Add a woocommerce_before_thankyou hook, Fix Issue/23537

### LINKED ISSUE #23537: thankyou hook
Can you please add do_action( 'woocommerce_before_thankyou') and do_action( 'woocommerce_after_thankyou') in checkout/thankyou.php just like you guys have in other templates such as cart, empty_cart, checkout templates. Because for this moment there is not start/end action to add some content for this specific page only. Please, it is not so annoying for you :)

### PR DESCRIPTION
**Pull request by Mahjouba91**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## Changes proposed in this Pull Request:

Closes #23537 .

### CODE REVIEW

**Comment by Mahjouba91** [comment]
There is no point of adding a `woocommerce_after_thankyou` hook in my opinion because there are already two hooks at the end :
`
	<?php do_action( 'woocommerce_thankyou_' . $order->payment_method, $order->id ); ?>
	<?php do_action( 'woocommerce_thankyou', $order->id ); ?>
`

**Comment by Stas0238** [comment]
@Mahjouba91 But how about adding some custom banner for ex. or custom text above all the content on this page ? i don't think it is so hard to add it because it will give users more options to add some content only for this page. I want to add banner via shortcode but i should rewrite the whole file to do this at this time ? If i had a hook at the top of page i could not think about future updates of this file because i would add just a content which not breaking any of content inside. Thanks for understanding

File: templates/checkout/thankyou.php [hunk 1]
```diff
@@ -22,7 +22,9 @@
 
 <div class="woocommerce-order">
 
-	<?php if ( $order ) : ?>
+	<?php if ( $order ) : 
+
+	do_action( 'woocommerce_before_thankyou', $order->id ); ?>
```
Comment (claudiosanches) [inline templates/checkout/thankyou.php]: Incorrect indentation, also should use `$order->get_id()` instead.

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Thansk for your help, but there's a few things to fix.

File: templates/checkout/thankyou.php (hunk 1 above)
Comment (Mahjouba91) [reply templates/checkout/thankyou.php]: Done, thanks

**Comment by Stas0238** [comment]
@Mahjouba91 Thanks for implementing this feature.
I also left one more request https://github.com/woocommerce/woocommerce/issues/23545 and I see you are working good so if you can you may help to make those thinks working too. Thanks 👍

**Review by claudiosanches — APPROVED** [review]
Great work, thanks for your help!

**Pull request merged**